crates/ruff_python_formatter/src/verbatim.rs

```diff
@@ -0,0 +1,505 @@
+                    }
+                }
+
+                // Unreachable because the function exits as soon as it reaches the end of the suppression
+                // and it already starts in a suppressed range.
+                SuppressionComments::SuppressionStarts { .. } => unreachable!(),
+                SuppressionComments::Formatted { .. } => unreachable!(),
+            }
+        }
+
+        comments.mark_verbatim_node_comments_formatted(AnyNodeRef::from(statement));
+
+        for range in CommentRangeIter::in_suppression(comments.trailing_comments(statement), source)
+        {
+            match range {
+                SuppressionComments::Suppressed { comments } => {
+                    for comment in comments {
+                        comment.mark_formatted();
+                    }
+                }
+
+                SuppressionComments::SuppressionEnds {
+                    suppressed_comments,
+                    format_on_comment,
+                    formatted_comments,
+                    format_off_comment: new_format_off_comment,
+                } => {
+                    for comment in suppressed_comments {
+                        comment.mark_formatted();
+                    }
+
+                    write!(
+                        f,
+                        [
+                            verbatim_text(TextRange::new(
+                                format_off_comment.end(),
+                                format_on_comment.start()
+                            )),
+                            format_comment(format_on_comment),
+                            hard_line_break(),
+                            trailing_comments(formatted_comments),
+                        ]
+                    )?;
+
+                    format_on_comment.mark_formatted();
+
+                    if let Some(new_format_off_comment) = new_format_off_comment {
+                        format_off_comment = new_format_off_comment;
+                        format_comment(format_off_comment).fmt(f)?;
+                        format_off_comment.mark_formatted();
+                    } else {
+                        return Ok(statement.statement());
+                    }
+                }
+
+                // Unreachable because the function exits as soon as it reaches the end of the suppression
+                // and it already starts in a suppressed range.
+                SuppressionComments::SuppressionStarts { .. } => unreachable!(),
+                SuppressionComments::Formatted { .. } => unreachable!(),
+            }
+        }
+
+        if let Some(next_statement) = statements.next() {
+            statement = SuiteChildStatement::Other(next_statement);
+            leading_node_comments = comments.leading_comments(next_statement);
+        } else {
+            let end = comments
+                .trailing_comments(statement)
+                .last()
+                .map_or(statement.end(), Ranged::end);
```

Comment:
nit: 
```suggestion
                .map(Ranged::end)
                .unwrap_or(statement.end());
```

Reply:
Clippy doesn't like that

```
warning: called `map(<f>).unwrap_or(<a>)` on an `Option` value. This can be done more directly by calling `map_or(<a>, <f>)` instead
   --> crates/ruff_python_formatter/src/verbatim.rs:327:23
    |
327 |               let end = comments
    |  _______________________^
328 | |                 .trailing_comments(statement)
329 | |                 .last()
330 | |                 .map(Ranged::end)
331 | |                 .unwrap_or(statement.end());
    | |___________________________________________^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#map_unwrap_or
    = note: `-W clippy::map-unwrap-or` implied by `-W clippy::pedantic`
help: use `map_or(<a>, <f>)` instead
    |
330 -                 .map(Ranged::end)
330 +                 .map_or(statement.end(), Ranged::end);
    |
```